src/App.tsx

```diff
@@ -1,61 +1,35 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+import { TODO } from './types/todo';
+import { NewTodo } from './components/NewTodo';
+import { initialTodos } from './services/initialTodos';
+import { getUserById } from './services/userById';
 
 export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
+  const [todos, setTodos] = useState<TODO[]>(initialTodos);
+  const maxId = Math.max.apply(
+    null,
+    todos.map(todo => todo.id),
+  );
 
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
+  const addTodo = ({ id, user, ...data }: TODO) => {
+    const newTodo = {
+      id: maxId + 1,
+      ...data,
+      user: getUserById(data.userId) || null,
```

Comment:
The `getUserById` function is used to fetch user data. If this function involves asynchronous operations (e.g., API calls), ensure it is handled properly with async/await or promises to avoid potential issues with undefined or null values.